src/EndevorController.ts

```diff
@@ -212,7 +213,15 @@ export class EndevorController {
         this.updateIDs(connName);
       } else {
         const endevorDataProvider = new EndevorDataProvider();
-        endevorDataProvider.addSession(connName);
+        try {
+          endevorDataProvider.addSession(connName);
+        } catch (error) {
+          const errMsg =
+            'You have an Endevor connection that refer to a non-existent profile named ' +
+            connName +
+            '. To resolve this, you can remove the connection with this profile name in your user settings.';
+          logger.warn(error.message, errMsg);
```

Comment:
> Will is be better to show the errMsg to the user and write the error message into a log for future analysis? How do you think?

Because this is the instruction that they told me to do. To display the message in the extension and output. I can also write in a log but I think this extension currently doesn't do that. @Alexandru-Dumitru am I right?

> why not to use multiline string to build the errMsg?
I don't mind. Can you show me how you want it to be coded?

Reply:
Yeah, sure, like you did with md syntax for coding or you can use this article: https://stackoverflow.com/questions/35225399/typescript-angular-multiline-string

Reply:
Actually, I think, logger.warn will do everything for you here. So all you need is to provide user message and output message (optional). First one will be written into user notification messages, and second one will be written in the output stream.

My initial point was about showing the user error message itself, and logging the pretty message, created here in the output. I suppose, we can just flip them in the logger call and it would be great to user and for developers.

Reply:
ok. so something like this?: `logger.warn(errMsg, error.message)`

Reply:
I assume, yes

Reply:
@nickImbirev Just for future reference, VSCode doesn't currently support newline characters in notifications. There's a debate about it [here](https://github.com/microsoft/vscode/issues/101589) but it sounds like they aren't interested in adding them...

I am swapping the messages so the user sees the pretty one. Will commit in a bit...